### PR TITLE
Notifications package

### PR DESCRIPTION
**Pull request by AMFTech512**

### CODE REVIEW

File: stream-projects/notifications/truffle.config.mjs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                    cronExpression
+                  }
+                }
+              }
+            `,
+            variables: {
+              input: {
+                resourcePath: `${PACKAGE}/_EventCron/notify-is-live`,
+                eventTopicPath: `${PACKAGE}/_EventTopic/notify-is-live`,
+                // TODO: switch this to 5 minutes
```
Comment (rileymiller) [inline stream-projects/notifications/truffle.config.mjs:146]: think every min is actually fine for this. users complain about the YT notifications taking too long to deliver

File: stream-projects/notifications/functions/src/notify-is-live.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    if (!channel) {
+      return makeResp(404, {
+        error: true,
+        message: "The channel requested from the org is undefined.",
+      });
+    }
+
+    // fetch `hasSentNotificationSinceLive` KV so we know if we need to send a notification
+    const hasSentKV = await fetchKV(
+      HAS_SENT_NOTIFICATION_KEY,
```
Comment (rileymiller) [inline stream-projects/notifications/functions/src/notify-is-live.ts:78]: does this need to be generated based on the current stream videoId or how does this reset from stream to stream?

Comment (AMFTech512) [reply stream-projects/notifications/functions/src/notify-is-live.ts:78]: If the function is triggered, and the stream is not live, then it resets the flag to false. The next time the stream goes live, it sees that the flag is false, sets it to true, and sends notifications to users. Then every subsequent time the function is triggered during the stream, it sees that the channel is live, but it also sees that the flag is true, so it doesn't send a notification to users.

By flag I mean `hasSentNotificationSinceLive`.

File: stream-projects/notifications/truffle.config.mjs (hunk 1 above)
Comment (AMFTech512) [reply stream-projects/notifications/truffle.config.mjs]: Update: https://github.com/trufflehq/truffle-packages/pull/49/commits/1e09c858a3d40f72f0e13c20efc241c6e7fa246d

**Pull request merged**